paramiko/client.py

```diff
@@ -518,6 +529,17 @@ def _auth(self, username, password, pkey, key_filenames, allow_agent,
         allowed_types = set()
         two_factor_types = set(['keyboard-interactive', 'password'])
 
+        # PKCS11 / Smartcard authentication
+        if username is not None and pkcs11pin is not None and \
+           pkcs11provider is not None:
+            try:
+                allowed_types = set(self._transport.auth_pkcs11(username,
+                                    pkcs11pin, pkcs11provider, pkcs11session))
+                if not two_factor:
```

Comment:
Yep. Small one, but could have caused issues.  The latest push should have resolved it.